src/services/facilityService.js

```diff
@@ -0,0 +1,88 @@
+
+/**
+ * A collection of methods that handles the database interactions
+ * for managing a Facility as an entity of the App.
+ *
+ * @class FacilityService
+ */
+class FacilityService {
+  /**
+   * Checks the room category value and creates a new record for any category added
+   * by user.
+   * @static
+   * @param {array} rooms - An array of rooms of a specific facility.
+   * @returns {Promise<Array>} A promise object with updated room properties.
+   * @memberof FacilityService
+   */
+  static async sortRoomCategory(rooms) {
+    const updatedRooms = rooms.map(async (room) => {
+      const { label } = await RoomCategory.findByPk(room.roomCategoryId);
+      if (label === 'Others') {
+        const { id } = await RoomCategory.create({ description: ' ', label: room.newCategory });
+        room.roomCategoryId = id;
+      }
+      return room;
+    });
+    return Promise.all(updatedRooms);
+  }
+
+  /**
+   * Creates an array of amenities for a specific facility.
+   * @static
+   * @param {object} amenities - Facility data to be recorded in the database.
+   * @param {number} id - Facility id.
+   * @returns {array}  An array of facility amenities.
+   * @memberof FacilityService
+   */
+  static sortFacilityAmenities(amenities, id) {
+    return amenities.map((item) => ({ amenityId: item, facilityId: id }));
+  }
+
+  /**
+   * Fetches a list of amenities returning an array of their labels.
+   * @static
+   * @param {array} amenityIds - Facility data to be recorded in the database.
+   * @returns {Promise<array>}  An array of facility amenities.
+   * @memberof FacilityService
+   */
+  static async findAmenitiesByIds(amenityIds) {
+    const amenityLabels = await Amenity.findAll({ where: { id: amenityIds }, attributes: ['label'] });
+    return amenityLabels.map(({ dataValues: { label } }) => label);
+  }
+
+
+  /**
+   * Creates a facility record in the database.
+   * @static
+   * @param {object} facilityInfo - Facility data to be recorded in the database.
+   * @returns {Promise<object>} A promise object with facility detail.
+   * @memberof FacilityService
+   */
+  static async createFacility(facilityInfo) {
+    const { amenities } = facilityInfo;
+    try {
+      const rooms = await FacilityService.sortRoomCategory(facilityInfo.rooms);
+      const result = await sequelize.transaction(async (t) => {
+        const facility = await Facility.create(facilityInfo, { transaction: t });
+        const updatedRooms = await updateCollection(rooms, { facilityId: facility.id });
+        const facilityAmenities = FacilityService.sortFacilityAmenities(amenities, facility.id);
+        await FacilityAmenity.bulkCreate(facilityAmenities, { transaction: t });
+        const roomsRecord = await Room.bulkCreate(updatedRooms, { transaction: t });
+        const amenityLabels = await FacilityService.findAmenitiesByIds(amenities);
+        return [facility.dataValues, extractArrayRecords(roomsRecord), amenityLabels];
```

Comment:
This looks beautiful, however, I thought a bit more on the response being sent. for the sake of consistency, I decided to omit this line `const amenityLabels = await FacilityService.findAmenitiesByIds(amenities);` and instead made a call to fetch the Facility instance just created including both relationships (rooms and amenities). This is so that I have the same response object pattern for when I create a new facility (using this function) and when I query for a single facility.

Reply:
Good catch @King-Success. I would update the PR as suggested.

Reply:
done @King-Success